src/App.js

```diff
@@ -35,9 +35,25 @@ function Navbar(props) {
 
 function NavItem(props) {
   const [open, setOpen] = useState(false);
+  const NavItemRef = useRef();
+
+  useEffect(() => {
+    document.addEventListener("click", handleClickOutside, true);
```

Comment:
Using the `mousedown` event instead of `click` will capture both left and right mouse clicks for anyone in need of that.